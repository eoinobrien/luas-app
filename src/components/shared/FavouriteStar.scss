button.favourite-star {
  background: none;
  border: none;
  margin-left: auto;

  svg {
    color: var(--secondary-text-colour);
    width: 2rem;
  }
}
