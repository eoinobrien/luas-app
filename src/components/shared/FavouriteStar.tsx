import React, { ReactElement } from 'react';
import './FavouriteStar.scss';
import { ReactComponent as Star } from './star.svg';
import { ReactComponent as StarFill } from './star-fill.svg';

interface FavouriteStarProps {
  favouriteClick: any;
  isFavourite: boolean;
  name: string;
}

const FavouriteStar: React.FC<FavouriteStarProps> = (props: FavouriteStarProps) => {
  return (
    <button
      className="favourite-star"
      aria-label={"Favourite " + props.name}
      onClick={() => props.favouriteClick()}>
      {props.isFavourite ? <StarFill style={{color: '#fdd835'}} /> : <Star />}
    </button>
  );
}

export default FavouriteStar;
