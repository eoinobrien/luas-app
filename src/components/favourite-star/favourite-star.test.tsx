import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import FavouriteStar from './favourite-star';

test('renders without crashing', () => {
    const mockFavouriteClick = jest.fn(x => console.log("clicked"));

    shallow(<FavouriteStar isFavourite={false} favouriteClick={mockFavouriteClick} name="name" />);
    shallow(<FavouriteStar isFavourite={true} favouriteClick={mockFavouriteClick} name="name" />);
});

test('FavouriteStar > Matches snapshot', () => {
    const mockFavouriteClick = jest.fn(x => console.log("clicked"));

    let tree = renderer
        .create(<FavouriteStar isFavourite={false} favouriteClick={mockFavouriteClick} name="station name" />)
        .toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer
        .create(<FavouriteStar isFavourite={true} favouriteClick={mockFavouriteClick} name="station name" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('FavouriteStar > Styles are correct', () => {
    let isFavourite = false;
    const mockFavouriteClick = jest.fn(() => isFavourite = !isFavourite);

    let star = mount(<FavouriteStar isFavourite={isFavourite} favouriteClick={mockFavouriteClick} name="station name" />);

    expect(star.find('svg').prop('style')).toBeUndefined();
    star.simulate('click');
    expect(mockFavouriteClick.mock.calls.length).toBe(1);

    star = mount(<FavouriteStar isFavourite={isFavourite} favouriteClick={mockFavouriteClick} name="station name" />);

    expect(star.find('svg').prop('style')).toHaveProperty('color', '#fdd835');
    star.simulate('click');
    expect(mockFavouriteClick.mock.calls.length).toBe(2);
});
