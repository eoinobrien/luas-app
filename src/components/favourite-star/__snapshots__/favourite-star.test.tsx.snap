// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteStar > Matches snapshot 1`] = `
<button
  aria-label="Favourite station name"
  className="favourite-star"
  onClick={[Function]}
>
  <svg>
    star.svg
  </svg>
</button>
`;

exports[`FavouriteStar > Matches snapshot 2`] = `
<button
  aria-label="Favourite station name"
  className="favourite-star"
  onClick={[Function]}
>
  <svg
    style={
      Object {
        "color": "#fdd835",
      }
    }
  >
    star-fill.svg
  </svg>
</button>
`;
